api/kyverno/v2beta1/spec_types.go

```diff
@@ -237,14 +233,28 @@ func (s *Spec) IsGenerateExisting() bool {
 }
 
 // GetFailurePolicy returns the failure policy to be applied
-func (s *Spec) GetFailurePolicy() kyvernov1.FailurePolicyType {
-	if s.FailurePolicy == nil {
-		return kyvernov1.Fail
+func (s *Spec) GetFailurePolicy(ctx context.Context) kyvernov1.FailurePolicyType {
+	if toggle.FromContext(ctx).ForceFailurePolicyIgnore() {
+		return kyvernov1.Ignore
+	} else if s.WebhookConfiguration != nil && s.WebhookConfiguration.FailurePolicy != nil {
+		return *s.WebhookConfiguration.FailurePolicy
+	} else if s.FailurePolicy != nil {
+		return *s.FailurePolicy
 	}
-	return *s.FailurePolicy
+	return kyvernov1.Fail
 }
 
-// GetFailurePolicy returns the failure policy to be applied
+func (s *Spec) GetWebhookTimeoutSeconds() *int32 {
```

Comment:
Same question about duplication.

Reply:
We could avoid the duplication if the function is implemented for the `WebhookConfiguration` struct as v2beta1 makes use of the same struct as v1. But, we still need to fetch the `webhookTimeoutSeconds` under the spec level if it exists.